adoptingerlang.org

```diff
@@ -4101,20 +4101,118 @@ With these broad lines covered, we can start exploring the processes and practic
+
+Then we have the complaint about the lone developer not caring about what the organisation does or knows, and who starts shipping things in a language nobody else knows. That developer might be doing this as part of resume padding, and then leave everyone behind after hopping to a new job. There's no protection against that, even when you mandate a single language. Hell, a developer could do the same by using your company's main language but by trying a new framework or new paradigm supported by the language or one of its libraries. Having a single unsupported [[https://aws.amazon.com/lambda/][Lambda]] could be trickier than a new language within a known deployment context.
+
+We mostly consider this negative behaviour, letting such work take place in complete isolation, a failure of process that must be addressed, rather than blaming the tool for being used.
+
+Finally, there's one major argument to undo. It's a commonly accepted idea that having many languages creates barriers, and that you will have an easier time moving developers around if you use one language and one framework. In fact, we'll call this the biggest myth against polyglot organisations.
+
+It's misguided because it assumes developers are like mechanical parts. As long as the interface is the same, they can be interchangeable. It ignores everything not related to the language. Different teams have different habits and dynamics you need to get used to. They work on projects with different histories and challenges, in possibly distinct domains, with different focuses and compromises done in how they engineer things. To add to this, even within a single language, there is a cost of migration: different frameworks, libraries, toolchains, or versions of these toolchains all require some adaptation.
+
+We argue that those altogether, represent a challenge that is often as significant—if not more—than learning a new programming language. Even if it's significant, the cost of learning new languages is less than it seems, mostly because it encompasses all of these underlying issues, which may still exist within a single language. It looks like the cost is high _because_ we traditionally don't provide support for any of the challenging things and they must be learned from scratch by switching ecosystems or teams.
```

Comment:
```suggestion
We argue that together, these factors represent a challenge that is often as significant—if not more so—than learning a new programming language. Even if it's significant, the cost of learning new languages is less than it seems, mostly because it encompasses all of these underlying issues, which may still exist within a single language. It looks like the cost is high _because_ we traditionally don't provide support for any of the challenging things and they must be learned from scratch by switching ecosystems or teams.
```